### ISSUE TITLE
make client less opionated

### ISSUE THREAD

**Issue by pmlopes** [vert-x3/vertx-redis-client#68]
Currently the client will behind the scenes enqueue request until it manages to connect to the server, it will try to reconnect on error and hold messages that have not been processed and attempt to handle them once a connection is available. It will also switch from pub/sub mode to normal mode and establish 2 connections so regular calls do not interfere with each other.

Although this is quite nice for people who just want it to work, for the case where one wants total control of what is going on this is bad decision.

I'd propose to create a different client class that is just purely simple. Establish a connection and has several handlers, for exception, error, etc, and sends commands.

**Comment by jmaasing** [comment]
It is also not currently possible to set Net client options such as idle timeout. Meaning if redis hangs the client hangs. It would be useful if the user of the API could have control over the connection settings.

**Comment by vietj** [comment]
so you mean something like a `RedisConnection` ?

**Comment by pmlopes** [comment]
Yes. Full control over the connection with handlers for connect, disconnect, exception plus using the core options for socket creation. And the only extra method would be one that could send a arbitrary command to redis and parse the response.

**Comment by vietj** [comment]
I would be in favor of this, see my other comment on the other issue (about having an SQLOperations like pattern for both connection/client).
